File: python/387. First Unique Character in a String/solution.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        freq = {}

        for char in s:
            freq[char] = freq.get(char, 0) + 1

        for i in range(len(s)):
            if freq[s[i]] == 1:
                return i

        return -1

File: java/387. First Unique Character in a String/Solution.java
class Solution {
    public int firstUniqChar(String s) {
        //s consists of only lowercase English letters. TIME -> O(n) SPACE -> O(1)

        int[] freq = new int[26];

        for (char c : s.toCharArray()) {
            freq[c - 'a']++;
        }

        for (int i = 0; i < s.length(); i++) {
            if (freq[s.charAt(i) - 'a'] == 1) return i;
        }

        return -1;
    }
}